doc/2/api/controllers/document/delete-by-query/index.md

```diff
@@ -94,11 +97,21 @@ Returns a `ids` array containing the list of deleted document identifiers.
   "action": "deleteByQuery",
   "requestId": "<unique request identifier>",
   "result": {
+    // Present only if 'source' parameter is not set, or set to false.
```

Comment:
I'm not comfortable to have two completely different response regarding the arguments.  
Also, we always returns array of documents in the API so maybe we can deprecate the `ids` response and returns an array of document with empty `_source` when the `source` parameter is not present.
What do you think?
(ping @scottinet )

Reply:
Yep, I was going to propose that, felt like having two different responses went against every principle, but i figured it was a breaking change.

Reply:
We can keep both the `ids` and `documents` arrays when the `source` option is not present so we avoid the breaking change.  
Then we deprecate the `ids` array in the documentation so people know they should stop using it

Reply:
Agreed :+1: